src/middleware/data-factory.ts

```diff
@@ -0,0 +1,16 @@
+import {
+  DefaultContext, DefaultState, Middleware, Next,
+} from 'koa';
+import { DataFactory } from 'rdf-js';
+
+export type DataFactoryContext<CustomT extends object = DefaultContext,
+  CustomF extends object = {}> = CustomT & { dataFactory: DataFactory & CustomF };
+
+export default <CustomF extends object>(dataFactory: DataFactory & CustomF):
```

Comment:
Trying to change the generic to `<CustomF extends DataFactory>`, but running in to a few problems in `data-factory.test.ts`.